createwindow/inputwindow.go

```diff
@@ -0,0 +1,61 @@
+package createwindow
+
+import (
+	"fmt"
+
+	"github.com/jezek/xgb"
+	"github.com/jezek/xgb/xproto"
+)
+
+// CreateInputWindow creates window to capture keycodes
+func (s *Session) CreateInputWindow(layout Layout, X *xgb.Conn, screenInfo *xproto.ScreenInfo) (i InputWindow, err error) {
+	wid, _ := xproto.NewWindowId(X)
+	cookie := xproto.CreateWindowChecked(X, screenInfo.RootDepth, wid, screenInfo.Root,
+		0, 0, uint16(layout.w), uint16(layout.h), 0,
+		xproto.WindowClassInputOutput, screenInfo.RootVisual,
+		xproto.CwBackPixel|xproto.CwEventMask,
+		[]uint32{
+			0xffffffff,
+			xproto.EventMaskEnterWindow |
+				xproto.EventMaskLeaveWindow |
+				xproto.EventMaskKeyPress |
+				xproto.EventMaskKeyRelease |
+				xproto.EventMaskStructureNotify})
+
+	if err := cookie.Check(); err != nil {
+		return nil, err
+	}
+
+	xproto.MapWindow(X, wid)
+	xproto.ConfigureWindow(X, wid,
+		xproto.ConfigWindowX|xproto.ConfigWindowY,
+		[]uint32{
+			uint32(layout.x), uint32(layout.y),
+		})
+
```

Comment:
combine this with createWindowChecked itself (?)

Reply:
this was added separately cause doing it in one go was creating the window at 0,0 irrespective of the x and y given (it's weird)